### ISSUE TITLE
Reset All Metadata - backend malfunction until complete

### ISSUE THREAD

**Issue by brendanhoar** [mcmonkeyprojects/SwarmUI#276]
### Expected Behavior

Metadata would be rebuilt in a temporary space, and then, when finished, swapped into place in a single transaction or equivalent, leading to no interruption of queued requests.

### Actual Behavior

Instead, metadata is removed before the scan finishes (perhaps before it starts), leading to interruptions or non-applied models in currently-queued requests. This is particularly problematic for libraries on slow media and/or large libraries

### Steps to Reproduce

Queue up a dozen or so image requests using multiple LORAs.
Press Reset All Metadata.
A pop-up error may notify you of a missing model that you actually have.
Back end may continue processing image request, however, some may be missing Loras.

### Debug Logs

N/A

### Other

This may be working as intended, but if so, it may be wise to warn the user that it should not be run during image generation and/or make the two processes mutually exclusive.

**Comment by mcmonkey4eva** [comment]
Yeah that's basically intended. The reset button is designed as a "something's gone wrong, tactical nuke all caches and figure things out from scratch".
There should however be a blocker in place to avoid anything failing from lack of data while reloading

**Comment by mcmonkey4eva** [comment]
added a protection
